tests/test_lower_exprs.py

```diff
@@ -181,3 +183,20 @@ def test_nocollection_staggered(self):
         leq = collect_derivatives.func([eq])[0]
 
         assert eq == leq
+
+    def test_lower_func_as_ind(self):
```

Comment:
You should create another class because this is specifically about the compiler pass "collect_derivatives". Since this is a test for a pice of code that is executed unconditionally, I'd put it into a "TestLowering" class at the top of the file